src/applications/static-pages/facilities/vet-center/NearByVALocations.jsx

```diff
@@ -0,0 +1,235 @@
+          ),
+        );
+        dispatch(
+          multiTypeQuery(
+            'VetCenter',
+            genQuery(boundingBox, coordinates, 'vet_center', true),
+          ),
+        );
+      }
+    },
+    [props, dispatch],
+  );
+
+  useEffect(
+    () => {
+      if (!props.togglesLoading && !isStartedLoading(props.multiLoading)) {
+        fetchNearbyVALocations();
+      }
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [props.togglesLoading, props.multiLoading],
+  );
+
+  useEffect(
+    () => {
+      const noDistancesToMeasure =
+        originalCoordinates.length === 0 || !isFinishedLoading(props);
+
+      if (nearbyVADistances || noDistancesToMeasure) {
+        return false;
+      }
+
+      const facilityCoordinates = joinMultiData(props)
+        .filter(center => center.id !== props.mainFacilityApiId)
+        .map(center => ({
+          id: center.id,
+          coordinates: [center.attributes.long, center.attributes.lat],
+        }));
+
+      const fetchDrivingData = async () => {
+        if (nearbyVADistances) {
+          return;
+        }
+        const response = await fetch(
+          distancesToNearbyVetCenters(
+            originalCoordinates,
+            facilityCoordinates.map(center => center.coordinates),
+          ),
+        );
+        const data = await response.json();
+        const nearbyDistances = data.distances.map(distance =>
+          convertMetersToMiles(distance[0]),
+        );
+
+        const facilityCoordinatesWithDistances = [...facilityCoordinates].map(
+          (center, index) => ({
+            ...center,
+            distance: nearbyDistances[index],
+          }),
+        );
+
+        setNearbyVADistances(facilityCoordinatesWithDistances);
+      };
+
+      fetchDrivingData();
+      return false;
+    },
+    [props, originalCoordinates, nearbyVADistances],
+  );
+  const normalizeFetchedFacilityProperties = useCallback(
+    vc => {
+      let centerDistance = false;
+
+      if (isFinishedLoading(props) && nearbyVADistances.length) {
+        const facilityDistance = nearbyVADistances.find(
+          distance => distance.id === vc.id,
+        );
+        centerDistance = facilityDistance.distance;
+      }
+      return buildFacility(vc, centerDistance);
+    },
+    [nearbyVADistances, props],
+  );
+
+  const normalizeFetchedFacilities = vcs => {
+    return vcs
+      .map(vc => normalizeFetchedFacilityProperties(vc))
+      .sort((a, b) => a.distance - b.distance)
+      .reduce(
+        (acc, vaf) => {
+          if (vaf.source === 'Health' && acc[0] === null) {
+            acc[0] = vaf;
+          } else if (vaf.source === 'VetCenter' && acc[1] === null) {
+            acc[1] = vaf;
+          } else if (vaf.source === 'Cemetery' && acc[2] === null) {
+            acc[2] = vaf;
+          }
+          return acc;
+        },
+        [null, null, null],
+      )
+      .filter(v => v);
```

Comment:
**Question:** What is this `filter` function for? It doesn't look like it would do anything.

Reply:
Nope, it does the job where if there is no valid facility for one of the types (i.e. no Health center, VetCenter, or Cemetery that gets put in the array that we don't have a random null in the list (since the reduce starts with three nulls). However, since the array uses placement for identifying the type of facility it needs to remove possible nulls.

Adding this as a comment